docs/developer/architecture/kibana-platform-plugin-api.asciidoc

```diff
@@ -0,0 +1,348 @@
+an HTTP route handler, a plugin just accesses it off of the first:
+argument:
+
+[source, typescript]
+----
+import type { CoreSetup } from 'kibana/server';
+
+export class MyPlugin {
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+    // handler is called when '/path' resource is requested with `GET` method
+    router.get({ path: '/path', validate: false }, (context, req, res) => res.ok({ content: 'ok' }));
+  }
+}
+----
+
+Different service interfaces can and will be passed to `setup`, `start`, and
+`stop` because certain functionality makes sense in the context of a
+running plugin while other types of functionality may have restrictions
+or may only make sense in the context of a plugin that is stopping.
+
+For example, the `stop` function in the browser gets invoked as part of
+the `window.onbeforeunload` event, which means you can’t necessarily
+execute asynchronous code here reliably. For that reason,
+`core` likely wouldn’t provide any asynchronous functions to plugin
+`stop` functions in the browser.
+
+The current lifecycle function for all plugins will be executed before the next
+lifecycle starts. That is to say that all `setup` functions are executed before
+any `start` functions are executed. Core services that expose functionality to
+plugins always have their lifecycle functions executed before any plugins.
+
+These are the contracts exposed by the core services for each lifecycle
+event:
+
+[cols=",,",options="header",]
+|===
+|lifecycle event |server contract|browser contract
+|_contructor_
+|{kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.plugininitializercontext.md[PluginInitializerContext]
+|{kib-repo}blob/{branch}/docs/development/core/public/kibana-plugin-core-public.plugininitializercontext.md[PluginInitializerContext]
+
+|_setup_
+|{kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.coresetup.md[CoreSetup]
+|{kib-repo}blob/{branch}/docs/development/core/public/kibana-plugin-core-public.coresetup.md[CoreSetup]
+
+|_start_
+|{kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.corestart.md[CoreStart]
+|{kib-repo}blob/{branch}/docs/development/core/public/kibana-plugin-core-public.corestart.md[CoreStart]
+
+|_stop_ |
+|===
+
+=== Integrating with other plugins
+
+Plugins can expose public interfaces for other plugins to consume. Like
+`core`, those interfaces are bound to the lifecycle functions `setup`
+and/or `start`.
+
+Anything returned from `setup` or `start` will act as the interface, and
+while not a technical requirement, all first-party Elastic plugins
+will expose types for that interface as well. 3rd party plugins
+wishing to allow other plugins to integrate with it are also highly
+encouraged to expose types for their plugin interfaces.
+
+*foobar plugin.ts:*
+
+[source, typescript]
+----
+import type { Plugin } from 'kibana/server';
+export interface FoobarPluginSetup { // <1>
```

Comment:
```suggestion
export interface FoobarPluginSetup { <1>
```

In other places we don't put a comment `//` before the reference, I didn't check how this renders but it seems like we don't usually place a comment before our references inside code.

Reply:
it breaks code highlight in WebStorm, but let's keep it consistent